## Detection of single nucleotide polymorphisms in _Saccharomyces cerevisiae_
Author: Anita Brzoza

### Table of contents

  - [Introduction](#introduction)
  - [Data downloading](#data-downloading)
  - [Quality control and filtering](#quality-control-and-filtering)
  - [Mapping to the reference genome](#mapping-to-the-reference-genome)
  - [Detection and annotation of SNPs](#detection-and-annotation-of-snps)
  - [Conclusions](#conclusions)
  - [References](#references)


### Introduction

_Saccharomyces cerevisiae_ was the first eukaryote whose genome was sequenced in 1996. Moreover, because of its small genome, it has become a model organism for studying gene function, and a comprehensive knowledge of yeast genetics and biochemistry is important for tracking the molecular processes that cause functional and genetic differences among strains of these microorganisms. The phenotypic variation is defined by many differernt types of variants, starting from single nucleotide polymorphisms (SNPs), insertions or deletions, etc. to large structural changes such as gene copy number variations (CNVs) and chromosomal translocations [(1)](#1-basile-a-de-pascale-f-bianca-f-rossi-a-frizzarin-m-de-bernardini-n--campanaro-s-2021-large-scale-sequencing-and-comparative-analysis-of-oenological-saccharomyces-cerevisiae-strains-supported-by-nanopore-refinement-of-key-genomes-food-microbiology-97-103753-doi101016jfm2021103753). 

In this study, the analysis will focus on the detection of single nucleotide polymorphisms in vac6 (wild type and mutant with impaired vacuole inheritance) and vac22 (wild type) _Saccharomyces cerevisiae_ strains. The data used are from the bioproject number [SRP003355](https://trace.ncbi.nlm.nih.gov/Traces/sra/?study=SRP003355) published on 2010-10-12 in the [National Center for Biotechnology Information Sequence Read Archive](https://www.ncbi.nlm.nih.gov/sra/) [(2)](#2-birkeland-s-r-jin-n-ozdemir-a-c-lyons-r-h-jr-weisman-l-s-wilson-t-e-2010-discovery-of-mutations-in-saccharomyces-cerevisiae-by-pooled-linkage-analysis-and-whole-genome-sequencing-genetics-1864-11271137-doiorg101534genetics110123232).

### Data downloading 



### Quality control and filtering

### Mapping to the reference genome

### Detection and annotation of SNPs

### Conclusions

### References

###### [1] Basile, A., De Pascale, F., Bianca, F., Rossi, A., Frizzarin, M., De Bernardini, N., … Campanaro, S. (2021). Large-scale sequencing and comparative analysis of oenological Saccharomyces cerevisiae strains supported by nanopore refinement of key genomes. Food Microbiology, 97, 103753. doi:10.1016/j.fm.2021.103753 

###### [2] Birkeland, S. R., Jin, N., Ozdemir, A. C., Lyons, R. H., Jr, Weisman, L. S., Wilson, T. E. (2010). Discovery of mutations in Saccharomyces cerevisiae by pooled linkage analysis and whole-genome sequencing. Genetics, 186(4), 1127–1137. doi.org/10.1534/genetics.110.123232

